src/browser.ts

```diff
@@ -0,0 +1,14 @@
+export class Browser {
+  constructor() {}
+  
+  fetch(url: string): void {
+    var xhr: XMLHttpRequest = new XMLHttpRequest();
```

Comment:
Let's wrap all of this code into a promise with `new Promise( (resolve, reject) => { ... } );`